client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -284,8 +286,13 @@ async function assignExistingCardProcessor( purchase, { storedDetailsId } ) {
 	} );
 }
 
-async function assignPayPalProcessor() {
-	// TODO: create a new paypal billing agreement
+async function assignPayPalProcessor( purchase ) {
+	return wpcomCreatePayPalAgreement( purchase.id, window.location.href, window.location.href ).then(
```

Comment:
Awesome! 👏  FWIW you can change this promise to return `makeRedirectResponse( data )` and it will do the redirect for you.

Reply:
Ah, thank you!  I threw these changes up in a hurry, but `makeRedirectResponse` looks like exactly what I should have been looking for.  I'm fixing that now.